src/AudioOutputULP.h

```diff
@@ -45,7 +51,7 @@ class AudioOutputULP : public AudioOutput
     bool waitingOddSample = true; // must be set to false for mono output
     int activeDACs = 3; // 1:DAC1; 2:DAC2; 3:both;
     bool stereoOutput = true;
-    const int opcodeCount = 18;
+    const int opcodeCount = 20;
```

Comment:
You can use a `sizeof()` operator (if an array then `sizeof(array) / sizeoff(array[0]` to get count of entries) here and avoid the hand fixup.

Reply:
I had seen that (I was planning to change it).
Actually I use that kind of expression in L 128 of the cpp file.
There is a problem.
Values in the array itself are defined based on that value (opcodeCount ...> indexAddress ...> bufferStart ...> 5th value of the array).  I believe that refactoring the code will require more thought (may be two passes defining the array). The code can be more elegant, easier to modify, with better variable naming, and more efficient (larger buffer for mono output), but without devoting more time you can not get those right. I prioritized fixing the errors and postponed refining the code.

Reply:
Fair enough.  Code itself is fine, no need to rewrite to make prettier!